src/main/kotlin/com/jetbrains/snakecharm/lang/parser/SmkStatementParsing.kt

```diff
@@ -289,13 +320,63 @@ class SmkStatementParsing(
         referenceMarker.drop()
         return false
     }
+
+    /**
+     * Parsing 'use' statement. Starts after first identifier and ends before the colon
+     */
+    private fun parseUseDeclaration(): Boolean {
+        val listOfRules = myBuilder.tokenType == PyTokenTypes.MULT
+        if (!listOfRules && myBuilder.tokenType != PyTokenTypes.IDENTIFIER) { // No rule name, no '*'
+            myBuilder.error(PyPsiBundle.message("PARSE.expected.symbols.first.quotation", "*", "rule name"))
+        } else if (!listOfRules) { // Have rule name, so mark it as reference
+            parseIdentifier()
+        } else { // Have '*'
+            nextToken()
+        }
+
+        if (myBuilder.tokenType != PyTokenTypes.AS_KEYWORD) {
+            checkMatches(PyTokenTypes.FROM_KEYWORD, PyPsiBundle.message("PARSE.0.expected", "from"))
```

Comment:
Maybe we need to introduce own token types for 'as', 'with' and remap python lexer token types to our own token types here, because PyCharm shows weird documentation (F1 / mouse hover) that is not relevant.

![image](https://user-images.githubusercontent.com/72933/127205026-9c655e88-8af7-448c-a253-016e91377909.png)

Reply:
P.S: we cannot defined 'with' and 'as' in lexer, because it will affect python code parsing, e.g. in cases:
```
with open("fff") as f:
   pass
   
 ```